src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -884,6 +885,30 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         RsDiagnostic.LoopOnlyKeywordUsedOutsideOfLoopError(expr.operator).addToHolder(holder)
     }
 
+    private fun checkArrayType(holder: RsAnnotationHolder, o: RsArrayType) {
+        collectDiagnostics(holder, o)
+        val sizeExpr = o.expr
+        if (sizeExpr != null && o.arraySize == null) {
+            checkArraySizeExpr(holder, sizeExpr)
+        }
+    }
+    private fun checkArrayExpr(holder: RsAnnotationHolder, o: RsArrayExpr) {
+        val sizeExpr = o.sizeExpr
+        if (sizeExpr != null) {
+            checkArraySizeExpr(holder, sizeExpr)
+        }
+    }
+
+    private fun checkArraySizeExpr(holder: RsAnnotationHolder, sizeExpr: RsExpr) {
+        val nonConstExpr = sizeExpr.descendantsOfTypeOrSelf<RsPathExpr>().firstOrNull {
```

Comment:
Unfortenately, `firstOrNull` may hide some errors from user.

For example, for the following code the compiler produces two errors
```rust
fn main() {
    let b  = 123usize;
    let c = 123;
    let a = [1; b + c];
}
```

```
error[E0435]: attempt to use a non-constant value in a constant
 --> src/main.rs:4:17
  |
4 |     let a = [1; b + c];
  |                 ^ non-constant value

error[E0435]: attempt to use a non-constant value in a constant
 --> src/main.rs:4:21
  |
4 |     let a = [1; b + c];
  |                     ^ non-constant value

error: aborting due to 2 previous errors
```

So, let's better write some recursive function that will collect all top most hon constant expressions in array-size expression and highlight all these expressions with error annotation